baseapp/baseapp.go

```diff
@@ -606,6 +616,22 @@ func (app *BaseApp) cacheTxContext(ctx sdk.Context, txBytes []byte) (sdk.Context
 	return ctx.WithMultiStore(msCache), msCache
 }
 
+func (app *BaseApp) SetCometInfo(ci comet.Info) {
+	app.cometInfo = ci
+}
+
+func (app *BaseApp) SetHeaderInfo(hi header.Info) {
+	app.headerInfo = hi
+}
+
+func (app *BaseApp) GetHeaderInfo(ctx context.Context) header.Info {
+	return app.headerInfo
```

Comment:
Don't height based queries have a different height or is this always the latest height?

Reply:
i was thinking of the same, should this replace height based queries as well? this approach wouldn't work then. I can try something else but currently the feature requirements are not defined in the adr so its hard to hit a moving target